crates/ruff/src/rules/flake8_pyi/rules/duplicate_union_member.rs

```diff
@@ -1,89 +1,79 @@
 use rustc_hash::FxHashSet;
-use rustpython_parser::ast::{self, Expr, Operator, Ranged};
-
-use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
-use ruff_macros::{derive_message_formats, violation};
-use ruff_python_ast::comparable::ComparableExpr;
+use rustpython_parser::ast::{self, Expr, Ranged};
 
 use crate::checkers::ast::Checker;
 use crate::registry::AsRule;
+use crate::rules::flake8_pyi::helpers::traverse_union;
+use ruff_diagnostics::{AutofixKind, Diagnostic, Edit, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::comparable::ComparableExpr;
 
 #[violation]
 pub struct DuplicateUnionMember {
     duplicate_name: String,
 }
 
-impl AlwaysAutofixableViolation for DuplicateUnionMember {
+impl Violation for DuplicateUnionMember {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         format!("Duplicate union member `{}`", self.duplicate_name)
     }
 
-    fn autofix_title(&self) -> String {
-        format!("Remove duplicate union member `{}`", self.duplicate_name)
+    fn autofix_title(&self) -> Option<String> {
+        Some(format!(
+            "Remove duplicate union member `{}`",
+            self.duplicate_name
+        ))
     }
 }
 
 /// PYI016
-pub(crate) fn duplicate_union_member(checker: &mut Checker, expr: &Expr) {
-    let mut seen_nodes = FxHashSet::default();
-    traverse_union(&mut seen_nodes, checker, expr, None);
-}
+pub(crate) fn duplicate_union_member<'a>(checker: &mut Checker, expr: &'a Expr) {
+    let mut seen_nodes: std::collections::HashSet<ComparableExpr<'_>, _> = FxHashSet::default();
```

Comment:
I'd probably recommend importing `HashSet` from `std::collections` so that you can just do: `let mut seen_nodes: HashSet<...> = ...`.